app/src/main/java/upday/droidconmvvm/MainViewModel.java

```diff
@@ -0,0 +1,43 @@
+package upday.droidconmvvm;
+
+import android.support.annotation.NonNull;
+
+import java.util.List;
+
+import rx.Observable;
+import rx.subjects.BehaviorSubject;
+import upday.droidconmvvm.datamodel.IDataModel;
+import upday.droidconmvvm.model.Language;
+
+/**
+ * View model for the main activity.
+ */
+public class MainViewModel {
+
+    @NonNull
+    private final IDataModel mDataModel;
+
+    @NonNull
+    private final BehaviorSubject<Language> mSelectedLanguage = BehaviorSubject.create();
+
+    public MainViewModel(@NonNull final IDataModel dataModel) {
+        mDataModel = dataModel;
+    }
+
+    @NonNull
+    public Observable<String> getGreeting() {
```

Comment:
Do you want to introduce our naming conventions for streams? Or wanna keep it simple?